erpnext/utilities/transaction_base.py

```diff
@@ -29,7 +29,13 @@ def validate_posting_time(self):
 			except ValueError:
 				frappe.throw(_('Invalid Posting Time'))
 
+		self.validate_future_posting()
 		self.validate_with_last_transaction_posting_time()
+	
+	def validate_future_posting(self):
```

Comment:
This will also prevent future posting of pure accounting transactions like Sales Invoice with no update stock. Future posting for pure accounting transaction should be allowed